node/self_out_response.js

```diff
@@ -85,8 +85,12 @@ function passError(codeString, message) {
         originalId: self.id,
         message: message
     });
-    self.inreq.outreq.errorEvent.emit(self, err);
     if (!self.closing) self.conn.lastTimeoutTime = 0;
+    process.nextTick(emitError);
+
+    function emitError() {
```

Comment:
defer should not happen here; its the wrong place; let me defer in the correct place.